src/ApparentHorizons/Expansion.cpp

```diff
@@ -0,0 +1,103 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ApparentHorizons/Expansion.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/MakeWithValue.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+
+template <typename Frame>
+DataVector expansion(
+    const tnsr::i<DataVector, 3, Frame>& r_hat, const DataVector& radius,
+    const tnsr::i<DataVector, 3, Frame>& normal_one_form,
+    const tnsr::ii<DataVector, 3, Frame>& d2x_radius,
+    const tnsr::II<DataVector, 3, Frame>& upper_spatial_metric,
+    const tnsr::ii<DataVector, 3, Frame>& extrinsic_curvature,
+    const tnsr::Ijj<DataVector, 3, Frame>& christoffel_2nd_kind) noexcept {
+  // Form unit normal spatial one-form to surface
+  auto surf_norm_one_form = normal_one_form;
+  const DataVector one_over_magnitude =
+      1.0 / magnitude(normal_one_form, upper_spatial_metric);
+  for (size_t i = 0; i < 3; ++i) {
+    surf_norm_one_form.get(i) *= one_over_magnitude;
+  }
+
+  // Form unit normal spatial vector to surface
+  auto surf_norm_vector =
```

Comment:
Since this function only works for `DataVector`s you can just use the constructor of Tensor, unless it would be useful to have the function be available for doubles